### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by danielochuba**
# Blog-App: Controllers specs
### In this pull request I did the following
- add test using `Rspec`
- Test following controllers
     - `Users`
     - `posts`

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    get user_url(user)
+    expect(response.body).to include("User's Name")
+    expect(response.body).to include('button to see all post')
+  end
+
+  it 'includes correct placeholder text in the response body' do
+    user = User.create!(valid_attributes)
+    get user_url(user)
+    expect(response.body).to include('Number of comments')
+  end
```
Comment (thecodechaser) [inline spec/requests/users_spec.rb:44]: - [ ] You are checking the tests for the `users#index` action but you should also check for the `users#show` action as requested in the project requirements. Please make sure to add those tests as well.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    post = Post.create!(valid_attributes)
+    get user_posts_path(post)
+    expect(response).to render_template('posts/index')
+  end
+
+  it 'includes correct placeholder text in the response body' do
+    post = Post.create!(valid_attributes)
+    get user_posts_path(post)
+    expect(response.body).to include('Number of comments on each post')
+  end
```
Comment (thecodechaser) [inline spec/requests/posts_spec.rb:39]: - [ ] You are checking the tests for the `posts#index` action but you should also check for the `posts#show` action as requested in the project requirements. Please make sure to add those tests as well.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @danielochuba👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply spec/requests/posts_spec.rb:39]: Fixed.

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply spec/requests/users_spec.rb:44]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @danielochuba 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Comment by danielochuba** [comment]
@thecodechaser @Abenezer-Tilahun Thank you all

**Pull request merged**